aw-server/src/endpoints/settings.rs

```diff
@@ -0,0 +1,84 @@
+use crate::endpoints::ServerState;
+use rocket::http::Status;
+use rocket::State;
+use rocket_contrib::json::Json;
+use std::sync::MutexGuard;
+
+use aw_datastore::{Datastore, DatastoreError};
+use aw_models::{Key, KV, KeyValue};
+
+fn parse_key(key: String) -> Result<String, Status> {
+    let namespace: String = "settings.".to_string();
+    if key.len() >= 128 {
+        return Err(Status::BadRequest);
+    } else {
+        Ok(namespace + key.as_str())
+    }
+}
+
+#[post("/", data = "<message>")]
+pub fn setting_set(state: State<ServerState>, message: Json<KV>) -> Result<Status, Status> {
+    let data = message.into_inner();
+
+    let setting_key = parse_key(data.key)?;
+
+    let datastore: MutexGuard<'_, Datastore> = endpoints_get_lock!(state.datastore);
+    let result = datastore.insert_key_value(&setting_key, &data.value);
+    return match result {
+        // TODO: Different status for replacement / creation (requires some sql adjustment)
```

Comment:
No need to do this I think

First of all I have no idea how to do it in sqlite in a performant way.
Secondly while there might possibly in the future be a use-case for checking this, you can just as well use the settings_list_get first to see if it will create or update. Not as fast, but should be good enough for most use-cases.

Reply:
Yeah makes sense. I'll remove the TODO, let's just do an issue / feature request if for some reason this becomes necessary in the future.

Reply:
fixed in 28f4c75